xds/server_options.go

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xds
+
+import (
+	"net"
+
+	"google.golang.org/grpc"
+)
+
+// Experimental
+//
+// Notice: All APIs in this file are EXPERIMENTAL and may be changed or removed
+// in a later release.
+
+// WithServingModeCallback returns a grpc.ServerOption which allows users to
+// register a callback to get notified about serving mode changes.
+//
+// NOTE: The returned ServerOption must *only* be used in a call to
+// xds.NewGRPCServer() and must not be used in a call to grpc.NewServer().
```

Comment:
To fix this: let's define an `EmptyServerOption` that we can embed instead.  That way we don't panic if it's passed to `grpc.NewServer` - it's just ignored.  And we don't need to filter it out from our list of options here, either.